internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_up_mgr.go

```diff
@@ -407,16 +407,21 @@ func (i *instanceTypeConsumptionSummaryCalculator) Calculate() (instanceTypeCons
 
 		if arrays.Contains(clusterStatesTowardReadyState, kafkaStreamingUnitCountPerCluster.Status) {
 			scaleUpActionIsOngoing = true
+			continue // ignore the rest of calculation as MaxUnits and Count are 0 for clusters that are in terraforming phase
+		}
+
+		clusterIsInDeletingState := arrays.Contains(clusterStatesTowardDeletion, kafkaStreamingUnitCountPerCluster.Status)
+		// ignore cluster is deleting state as they can't accept kafka anymore
+		if clusterIsInDeletingState {
+			continue
 		}
 
 		if kafkaStreamingUnitCountPerCluster.FreeStreamingUnits() >= int32(biggestKafkaInstanceSizeCapacityConsumption) {
```

Comment:
With this condition (without the added skip above), all the clusters in deleting states were considered for this evaluation. 
If one of them had a max size big enough to accommodate the biggest streaming unit, then we were considering that we've enough space in the region for that instance size, so no scale up is required. 
While in fact, we should omit any clusters in these states from capacity evaluation consideration as the clusters that in deleting states can't accept kafka anymore: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/b27cb8ccf683b2ce71cf1d36a4c9f2b561b355ad/internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_up_mgr.go#L368